src/cloudConsole/uploadFileToCloudShell.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { IActionContext, parseError } from "@microsoft/vscode-azext-utils";
+import { createReadStream } from "fs";
+import { basename } from "path";
+import { ProgressLocation, Uri, window, workspace } from "vscode";
+import { ext } from "../extensionVariables";
+import { localize } from "../utils/localize";
+import { OSName, OSes, createCloudConsole, shells } from "./cloudConsole";
+
+
+async function cloudConsole(os: OSName) {
+    const shell = createCloudConsole(await ext.subscriptionProviderFactory(), os);
+    if (shell) {
+        void shell.terminal?.then(terminal => terminal.show());
+        return shell;
+    }
+
+    return undefined;
+}
+
+
+export async function uploadFileToCloudShell(_context: IActionContext, uri?: Uri) {
+    (async () => {
+        if (!workspace.isTrusted) {
+            throw new Error(localize('uploadingRequiresTrustedWorkspace', 'File upload only works in a trusted workspace.'));
+        }
+        let shell = shells[0];
```

Comment:
Does this mean that you can have multiple local shell instances but they all are connected to the same remote instance?  I notice if I upload a file, they appear in all cloud shell terminals when I list contents

Reply:
Not exactly sure what you mean, but every user just has one cloud shell instance in Azure. So if you spin up multiple clients, the will all be connected to the same container.

Reply:
Okay cool, you covered what I was trying to confirm, I probably wasn't referring to each entity using the correct name which could make it confusing to understand.